src/main/java/org/jenkinsci/plugins/configfiles/custom/security/CustomizedCredentialMapping.java

```diff
@@ -0,0 +1,117 @@
+ OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ THE SOFTWARE.
+ */
+package org.jenkinsci.plugins.configfiles.custom.security;
+
+import com.cloudbees.plugins.credentials.common.IdCredentials;
+import com.cloudbees.plugins.credentials.common.StandardListBoxModel;
+import com.cloudbees.plugins.credentials.common.StandardUsernameCredentials;
+import com.cloudbees.plugins.credentials.CredentialsMatchers;
+import com.cloudbees.plugins.credentials.CredentialsProvider;
+import hudson.Extension;
+import hudson.model.AbstractDescribableImpl;
+import hudson.model.Descriptor;
+import hudson.model.Item;
+import hudson.model.Queue;
+import hudson.model.queue.Tasks;
+import hudson.security.ACL;
+import hudson.util.ListBoxModel;
+import jenkins.model.Jenkins;
+import org.apache.commons.lang.StringUtils;
+import org.jenkinsci.plugins.plaincredentials.StringCredentials;
+import org.kohsuke.stapler.AncestorInPath;
+import org.kohsuke.stapler.DataBoundConstructor;
+import org.kohsuke.stapler.QueryParameter;
+
+import java.io.Serializable;
+import java.util.Collections;
+import java.util.List;
+
+public class CustomizedCredentialMapping extends AbstractDescribableImpl<CustomizedCredentialMapping> implements Serializable {
+
+    private final String tokenKey;
+    private final String credentialsId;
+
+    @DataBoundConstructor
+    public CustomizedCredentialMapping(String tokenKey, String credentialsId) {
+        this.tokenKey = tokenKey;
+        this.credentialsId = credentialsId;
+    }
+
+    public String getTokenKey() {
+        return tokenKey;
+    }
+
+    public String getCredentialsId() {
+        return credentialsId;
+    }
+
+    public Descriptor<CustomizedCredentialMapping> getDescriptor() {
+        return DESCRIPTOR;
+    }
+
+    private static final DescriptorImpl DESCRIPTOR = new DescriptorImpl();
+
+    @Extension
+    public static class DescriptorImpl extends Descriptor<CustomizedCredentialMapping> {
+
+        public ListBoxModel doFillCredentialsIdItems(
+                @AncestorInPath Item item,
+                @QueryParameter String credentialsId
+                ) {
+
+            StandardListBoxModel result = new StandardListBoxModel();
+            if (item == null) {
+                if (!Jenkins.get().hasPermission(Jenkins.ADMINISTER)) {
+                    return result.includeCurrentValue(credentialsId);
+                }
+            } else {
+                if (!item.hasPermission(Item.EXTENDED_READ)
+                    && !item.hasPermission(CredentialsProvider.USE_ITEM)) {
+                    return result.includeCurrentValue(credentialsId);
+                }
+            }
+
+            return result
+                .includeMatchingAs(
+                    item instanceof Queue.Task
+                        ? Tasks.getAuthenticationOf((Queue.Task) item)
+                        : ACL.SYSTEM,
+                    Jenkins.get(),
```

Comment:
```suggestion
                    item,
```

Reply:
Latest revision has this change in it now